dll/cpl/desk/desktop.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * PROJECT:     ReactOS Display Control Panel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Desktop customization property page
+ * COPYRIGHT:   Copyright 2018-2022 Stanislav Motylkov <x86corez@gmail.com>
+ */
+
+#include "desk.h"
+
+#define FCIDM_DESKBROWSER_REFRESH    0xA220  /* From shresdef.h */
+
+static const TCHAR szHideDesktopIcons[] = TEXT("Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\HideDesktopIcons");
+static const TCHAR szClassicStartMenu[] = TEXT("ClassicStartMenu");
+static const TCHAR szNewStartPanel[] = TEXT("NewStartPanel");
+
+struct
+{
+    LPCTSTR CLSID;
+    UINT Checkbox;
+} DesktopIcons[] = {
+    /* Number of elements must be NUM_DESKTOP_ICONS */
+    {TEXT("{450D8FBA-AD25-11D0-98A8-0800361B1103}"), IDC_ICONS_MYDOCS},   /* My Documents */
+    {TEXT("{208D2C60-3AEA-1069-A2D7-08002B30309D}"), IDC_ICONS_MYNET},    /* My Network Places */
+    {TEXT("{20D04FE0-3AEA-1069-A2D8-08002B30309D}"), IDC_ICONS_MYCOMP},   /* My Computer */
+    {TEXT("{871C5380-42A0-1069-A2EA-08002B30309D}"), IDC_ICONS_INTERNET}, /* Internet Browser */
+};
+
+C_ASSERT(_countof(DesktopIcons) == NUM_DESKTOP_ICONS);
```

Comment:
What I usually do also in those cases, is to hardcode the expected size of the array, like in this example:
`DesktopIcoons[NUM_DESKTOP_ICONS]`.
So that if someone adds an extra line in it without changing the `NUM_DESKTOP_ICONS`, a compile error will be emitted.
Likewise, if `NUM_DESKTOP_ICONS` define value is changed: (i) if it's reduced without removing a corrresponding line from the array, you'll get a compile error, and (ii) if it's increased, you may get a compile warning telling that not all array elements are initialized.